lib/mayaUsd/sceneIndex/proxyShapeSceneIndexPlugin.cpp

```diff
@@ -97,10 +133,11 @@ namespace MAYAUSD_NS_DEF {
 ///////////////////////// MayaUsdProxyShapeSceneIndex
 
 MayaUsdProxyShapeSceneIndex::MayaUsdProxyShapeSceneIndex(
-    UsdImagingStageSceneIndexRefPtr inputSceneIndex,
+    HdSceneIndexBaseRefPtr          inputSceneIndex,
+    UsdImagingStageSceneIndexRefPtr usdImagingStageSceneIndex,
     MayaUsdProxyShapeBase*          proxyShape)
     : ParentClass(inputSceneIndex)
-    , _usdImagingStageSceneIndex(inputSceneIndex)
+    , _usdImagingStageSceneIndex(usdImagingStageSceneIndex)
```

Comment:
As far as I can tell, this is passed in as a argument to support a temporary workaround, and init on first use of _usdImagingStageSceneIndex.  However, why is it the responsibility of the MayaUsdProxyShapeSceneIndex to set the stage in Populate() of something that is arguably external to it?  Can't we set the stage onto the _usdImagingStageSceneIndex upon its construction?  A proxy shape node's stage won't change.

Reply:
No the MayaUsdProxyShapeSceneIndex also listens for changes:
```
void MayaUsdProxyShapeSceneIndex::ObjectsChanged(
    const MayaUsdProxyStageObjectsChangedNotice& notice)
{
    // MAYA-126804 TODO: This should be enough to trigger dirty notifications being send
    // when a matrix is changed. However this is not working due missing
    // UsdImagingMeshAdapter::Invalidate implementation. This is not part of USD v22.11 however has
    // been fixed in later versions.
    _usdImagingStageSceneIndex->ApplyPendingUpdates();
}
```

Reply:
Also please note, that changing the inner working of MayaUsdProxySHapeSceneIndex is out of scope for these changes.